website/project/model.py

```diff
@@ -1213,6 +1233,51 @@ def nodes_primary(self):
         ]
 
     @property
+    def depth(self):
+        return len(self.parents)
+
+    def next_descendants(self, auth, condition=lambda auth, node: True):
+        """
+        Recursively find the first set of descedants under a given node that meet a given condition
+
+        returns a list of [(node, [children]), ...]
+        """
+        ret = []
```

Comment:
Instead of building the list in-memory, can this be method be made into a generator?